src/Controller/RequestDataHandler/ShopProductsSortDataHandler.php

```diff
@@ -12,54 +12,29 @@
 
 namespace BitBag\SyliusElasticsearchPlugin\Controller\RequestDataHandler;
 
-use BitBag\SyliusElasticsearchPlugin\Context\TaxonContextInterface;
 use BitBag\SyliusElasticsearchPlugin\PropertyNameResolver\ConcatedNameResolverInterface;
 use Sylius\Component\Channel\Context\ChannelContextInterface;
 use UnexpectedValueException;
 
 final class ShopProductsSortDataHandler implements SortDataHandlerInterface
 {
-    private ConcatedNameResolverInterface $channelPricingNameResolver;
-
-    private ChannelContextInterface $channelContext;
-
-    private TaxonContextInterface $taxonContext;
-
-    private ConcatedNameResolverInterface $taxonPositionNameResolver;
-
-    private string $soldUnitsProperty;
-
-    private string $createdAtProperty;
-
-    private string $pricePropertyPrefix;
-
     public function __construct(
-        ConcatedNameResolverInterface $channelPricingNameResolver,
-        ChannelContextInterface $channelContext,
-        TaxonContextInterface $taxonContext,
-        ConcatedNameResolverInterface $taxonPositionNameResolver,
-        string $soldUnitsProperty,
-        string $createdAtProperty,
-        string $pricePropertyPrefix
+        private ConcatedNameResolverInterface $channelPricingNameResolver,
+        private ChannelContextInterface $channelContext,
+        private string $soldUnitsProperty,
+        private string $createdAtProperty,
+        private string $pricePropertyPrefix
     ) {
-        $this->channelPricingNameResolver = $channelPricingNameResolver;
-        $this->channelContext = $channelContext;
-        $this->taxonContext = $taxonContext;
-        $this->taxonPositionNameResolver = $taxonPositionNameResolver;
-        $this->soldUnitsProperty = $soldUnitsProperty;
-        $this->createdAtProperty = $createdAtProperty;
-        $this->pricePropertyPrefix = $pricePropertyPrefix;
     }
 
     public function retrieveData(array $requestData): array
     {
         $data = [];
-        $positionSortingProperty = $this->getPositionSortingProperty();
 
-        $orderBy = $requestData[self::ORDER_BY_INDEX] ?? $positionSortingProperty;
+        $orderBy = $requestData[self::ORDER_BY_INDEX] ?? $this->createdAtProperty;
         $sort = $requestData[self::SORT_INDEX] ?? self::SORT_ASC_INDEX;
 
-        $availableSorters = [$positionSortingProperty, $this->soldUnitsProperty, $this->createdAtProperty, $this->pricePropertyPrefix];
+        $availableSorters = [$this->soldUnitsProperty, $this->createdAtProperty, $this->pricePropertyPrefix];
```

Comment:
What if someone want to add custom sorter? Class decoration will be necessary to do this? What do you think about changing this prefixes to one symfony array parameter which can be easily override?

Reply:
Custom sort logic has to be implemented anyway. Not in scope of this task, but thanks for the idea. We can maybe schedule it for some other time.